docs/src/dist-training-torch/ddp_intro.md

```diff
@@ -0,0 +1,63 @@
+
+*   **Lance Integration:** `lance.torch.data.SafeLanceDataset`
+*   **Best For:** Finite, indexable datasets. This covers the vast majority of standard use cases, such as image classification (ImageNet, FOOD101), text classification, or any scenario where the dataset is a discrete collection of samples.
+
+### Iterable-Style: `torch.utils.data.IterableDataset`
+An iterable-style dataset works like a Python generator. It implements the `__iter__(self)` method, which yields data items sequentially as a stream. It often has no known length and does not support random access to a specific index. This makes it ideal for handling data that doesn't fit the indexed collection model.
+
+*   **Lance Integration:** `lance.torch.data.LanceDataset`
+*   **Best For:** True streaming data sources (e.g., from a database cursor or real-time log), extremely large datasets where building a full index is impractical, or when implementing complex, custom sampling logic that goes beyond what standard PyTorch samplers offer. 
+
+## Choosing Your Data Loading Strategy
+
+For most projects, the choice is between the high-performance map-style pattern and the flexible but slower simple iterable-style pattern. The table below provides a detailed comparison to help beginners select the best approach for their needs.
+
+| Feature | Map-Style (`SafeLanceDataset`) | Simple Iterable-Style (`LanceDataset`) |
+| :--- | :--- | :--- |
+| **Primary Use Case** | Standard, indexable datasets (e.g., image classification) | Streaming data, very large datasets, or custom sampling logic. |
+| **Performance** | **Highest**. The recommended default for throughput. | **Lower**. I/O is serialized in a single process. |
+| **Parallelism** | **High**. Data decoding is parallelized across multiple workers. | **Low**. Data is fetched and decoded sequentially in a single stream. |
+| **Shuffling** | Natively supported via PyTorch samplers (e.g., `DistributedSampler`).  | Requires custom implementation (e.g., buffering data). |
+| **Epoch Length** | Deterministic. The dataset has a known length (`__len__`).  | Not deterministic by default. Requires manual loop control. |
+| **PyTorch `num_workers`** | `> 0` (Recommended for performance). | `0` (mutli-worker requires custom implementation). |
+| **Sampler** | PyTorch `DistributedSampler`. | Custom (E.g. Lance `ShardedBatchSampler`. ) |
+
+It is recommended to use map‑style datasets by default. Map‑style give you their size ahead of time, easier to shuffle, and allow for easy parallel loading. But if you're reading from a massive data source remotely and/or you need to use custom sampling logic, you should use iterable style dataloader. Common techniques and pitfalls of each approach is discussed in their respective sections.
```

Comment:
Reading this table I don't see any real reason to use an iterable style dataset for Lance?  Doesn't pytorch support a custom sampler for map datasets?

Reply:
Hmm yeah let me rephrase that a little.. The main intention of the intro part is to not be lance specific. We get questions like - whats the difference bw the different approaches and why should I use one over other? 
The general recommendation is that - Always use map-style unless you can't. Its probably better to simply say that lance does not materialize the entire dataset in memory, so you should always stick to Map-style